### ISSUE TITLE
PRP: Request CVE-2021-31805 Struts2 RCE S2-062

### ISSUE THREAD

**Issue by C4o** [google/tsunami-security-scanner-plugins#251]
Hi there,

I would like to contribute the implementation for a plugin that detects CVE-2021-31805, published on 12 Apr, aka S2-062. 
It's a bypass of CVE-2020-17530, some of the tag’s attributes could perform a double evaluation if a developer applied forced OGNL evaluation by using the %{...} syntax and using forced OGNL evaluation on untrusted user input can lead to a Remote Code Execution and security degradation..

Vulnerability details:

Score: 9.8 CRITICAL
Vector: CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H

References:
https://nvd.nist.gov/vuln/detail/CVE-2021-31805
https://github.com/advisories/GHSA-v8j6-6c2r-r27c
https://www.openwall.com/lists/oss-security/2022/04/12/6

The vulnerability should be remotely exploitable without authentication and user interaction. Yes.

The detector should provide a reliable false-positive free detection report. Yes.

The detection capability should be easy to verify using both vulnerable and fixed Docker images. Yes.

The vulnerability should have a relatively large impact radius. Yes.

Please let me know if this is in scope as I've already made the development .

Thanks,
C4o

**Comment by maoning** [comment]
@C4o do you know if the same payload also works for CVE-2020-17530? Can we update the existing plugin to cover both vulnerabilities?

**Comment by C4o** [comment]
@maoning yes, of course. Affected versions of CVE-2020-17530 are inclued by CVE-2021-31805.
CVE-2020-17530 aka s2-061 : Struts 2.0.0-Struts 2.5.25
CVE-2021-31805 aka s2-062 : Struts 2.0.0-Struts 2.5.29
